pkg/kubeconfig/kubeconfig.go

```diff
@@ -43,25 +43,28 @@ type configLoader struct {
 
 func newConfigLoader(existing *cluster.Cluster) (*configLoader, error) {
 	pubKeyPath := local.Expand(existing.SSH.PublicKeyPath)
-	// --------------------------------------------------------------------------------
-	//
-	// @kris-nova
-	//
-	// We don't need to check a key first, because SSH can support multiple auth implementations
-	// So here we just add BOTH a key based auth, and an arbitrary agent. Please don't touch this
-	// without talking to @kris-nova first OR unless something is just completely fucked
-	//
+
+	// create an agent, either a connection to the one of the environment of the user or
+	// (if there is none) a go-implementation of a simple keyring
 	sshAgent := agent.NewAgent()
-	sshAgentWithKey, err := sshAgent.AddKey(pubKeyPath)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to add key: %v", err)
-	}
 	sshConfig := &ssh.ClientConfig{
 		User:            existing.SSH.User,
 		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
 	}
 	sshConfig.Auth = append(sshConfig.Auth, sshAgent.GetAgent())
-	sshConfig.Auth = append(sshConfig.Auth, sshAgentWithKey.GetAgent())
```

Comment:
I think we need to add this second agent back at a minimum.

Reply:
I'm not sure why this second line is needed. In the old code the key was added to the "sshAgent" too **and** an new keyring was returned which was stored as `sshAgentWithKey`. The code
```go
sshAgent.AddKey(pubKeyPath)
``` 
adds the key to `sshAgent` and also returns a new keyring which contains the key too.

So the loaded key was stored in `sshAgent` and in a second keyring --> the loaded key was inserted twice, in two agents. i do not see, why this should be necessary